packages/workflows/src/helper/pipe.ts

```diff
@@ -53,6 +71,15 @@ export function pipe<T>(
   input: PipelineInput,
   ...functions: [...PipelineHandler[], PipelineHandler<T>]
 ): WorkflowStepHandler {
+  // Apply the aggregator just before the last handler
+  if (
+    (input.merge || input.mergeAlias || input.mergeFrom) &&
+    functions.length
+  ) {
+    const handler = functions.pop()!
```

Comment:
question: Why do we only want to merge before the last handler? My steps broke after merging in these changes, because my middlewares were expecting the merged data shape too.

Reply:
The idea was to merge everything including the middleware result as we discussed on slack no? Previously, when used manually you could have placed the aggregator at any position you wanted before or after middleware's are applied, you can decide tou not use the automatic merge and instead use it manually, it is also possible